### ISSUE TITLE
SIGSEGV when compiling `tf.keras.layers.ZeroPadding3D` with extremely large padding under XLA

### ISSUE THREAD

**Issue by tinywisdom** [tensorflow/tensorflow#116716]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

binary

### TensorFlow version

2.21.0

### Custom code

Yes

### OS platform and distribution

Ubuntu 22.04.4 LTS (x86_64)

### Mobile device

Ubuntu 22.04.4 LTS (x86_64)

### Python version

3.10.16

## Summary

`tf.keras.layers.ZeroPadding3D` with an extremely large `padding` value can crash the TensorFlow worker process with `SIGSEGV` when executed through `tf.function(jit_compile=True)` on CPU.

The failure is not raised as a Python exception. Instead, the subprocess exits with signal 11:

```text
worker died by SIGSEGV
exitcode: -11
```

This suggests a native runtime crash in the XLA/TensorFlow execution path, likely related to shape computation, integer overflow, or invalid buffer planning for the padded tensor.

### Standalone code to reproduce the issue

```shell
import os
import signal
import multiprocessing as mp

# Force CPU execution and reduce unrelated TensorFlow logs.
os.environ["TF_CPP_MIN_LOG_LEVEL"] = "2"
os.environ["CUDA_VISIBLE_DEVICES"] = "-1"


def worker():
    import tensorflow as tf

    tf.config.run_functions_eagerly(False)
    tf.keras.mixed_precision.set_global_policy("mixed_float16")

    tf.config.optimizer.set_experimental_options(
        {
            "arithmetic_optimization": True,
            "auto_mixed_precision": True,
            "constant_folding": True,
            "layout_optimizer": False,
            "remapping": True,
            "shape_optimization": True,
        }
    )

    tf.config.optimizer.set_jit(True)

    class MyModel(tf.keras.Model):
        def __init__(self):
            super().__init__()
            self.pad = tf.keras.layers.ZeroPadding3D(
                padding=1610612736
            )

        def call(self, x):
            return self.pad(x)

    model = MyModel()
    x = tf.random.uniform((1, 1, 2, 2, 3), dtype=tf.float32)

    @tf.function(
        jit_compile=True,
        experimental_relax_shapes=False,
    )
    def compiled_forward(inp):
        return model(inp)

    y = compiled_forward(x)

    # Usually unreachable if SIGSEGV is triggered.
    print(y.shape)


def main():
    proc = mp.get_context("spawn").Process(target=worker)
    proc.start()
    proc.join(120)

    if proc.is_alive():
        proc.terminate()
        proc.join()
        print("worker timeout")
        return 124

    exitcode = proc.exitcode

    if exitcode is not None and exitcode < 0:
        sig_num = -exitcode
        sig_name = signal.Signals(sig_num).name
        print(f"worker died by {sig_name}")
        print(f"exitcode: {exitcode}")
        return 128 + sig_num

    print(f"worker exited normally, exitcode={exitcode}")
    return exitcode or 0


if __name__ == "__main__":
    raise SystemExit(main())
```

### Relevant log output

```shell
worker died by SIGSEGV
exitcode: -11
```
